packages/lib-utils/src/types/api-discovery.ts

```diff
@@ -0,0 +1,19 @@
+import type { Store, AnyAction } from 'redux';
+import type { ActionType as Action } from 'typesafe-actions';
+import type { K8sVerb } from './k8s';
+
+export type InitAPIDiscovery = (store: Store<unknown, Action<AnyAction>>) => void;
```

Comment:
🤔 
I wonder if we want to take a Generic approach to the store data... extending our data. Not sure it matters though as we'll probably move away from Redux to React Context or something

Reply:
I don't personally think generics are a great idea here, though I'm open to suggestions.

Reply:
Let us not worry about it right now. The store is absolutely not `unknown` though.

Reply:
Tracked here: https://issues.redhat.com/browse/HAC-765

Reply:
Generics make sense if we or the consumer can make a reliable assumption about the types involved. If not, using `unknown` is perfectly fine.